pkgs/build-support/src-only/default.nix

```
@@ -1,22 +1,37 @@
-{ stdenv }:
+{ lib, stdenvNoCC }:
+
 # srcOnly is a utility builder that only fetches and unpacks the given `src`,
 # and optionally patching with `patches` or adding build inputs.
 #
 # It can be invoked directly, or be used to wrap an existing derivation. Eg:
 #
 # > srcOnly pkgs.hello
 #
+
 attrs:
 let
-  args = if builtins.hasAttr "drvAttrs" attrs then attrs.drvAttrs else attrs;
-  name = if builtins.hasAttr "name" args then args.name else "${args.pname}-${args.version}";
+  args = attrs.drvAttrs or attrs;
+  name = args.name or "${args.pname}-${args.version}";
+  drv = stdenvNoCC.mkDerivation (
+    args
+    // {
+      name = "${name}-source";
+
+      phases = [
+        "unpackPhase"
+        "patchPhase"
+        "installPhase"
+      ];
+      separateDebugInfo = false;
+
+      dontUnpack = false;
+
+      dontInstall = false;
+      installPhase = "cp -pr --reflink=auto -- . $out";
+    }
+  );
 in
-stdenv.mkDerivation (args // {
-  name = "${name}-source";
-  installPhase = "cp -pr --reflink=auto -- . $out";
-  outputs = [ "out" ];
-  separateDebugInfo = false;
-  dontUnpack = false;
-  dontInstall = false;
-  phases = ["unpackPhase" "patchPhase" "installPhase"];
-})
+if (args.dontUnpack or false) then
+  lib.warn "srcOnly: derivation has dontUnpack set, overriding"
+else
+  drv
```

`lib.warn` doesn't have enough parameters!

```suggestion
lib.warnIf (args.dontUnpack or false) "srcOnly: derivation has dontUnpack set, overriding" drv
```